src/Service/Context/PageContext.php

```diff
@@ -94,7 +95,9 @@ public function __construct(ConfigFactoryInterface $config_factory, EntityTypeMa
    */
   private function setPageContextCredential($credential_settings) {
     foreach (SELF::$CREDENTIAL_MAPPING as $credential_key => $tag_name) {
-      $this->pageContext[$tag_name] = $credential_settings[$credential_key];
+      if (isset($credential_settings[$credential_key]) {
+        $this->pageContext[$tag_name] = $credential_settings[$credential_key];
+      }
```

Comment:
It doesn't mind if they are optional (like this one).  It has checks in place to ensure that credentials are available, etc. (regardless of how they are added to the page).